src/Admin/AdminExtensionInterface.php

```diff
@@ -12,29 +12,34 @@
 namespace Sonata\AdminBundle\Admin;
 
 use Knp\Menu\ItemInterface as MenuItemInterface;
-use Sonata\AdminBundle\Datagrid\DatagridMapper;
-use Sonata\AdminBundle\Datagrid\ListMapper;
-use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
-use Sonata\AdminBundle\Form\FormMapper;
-use Sonata\AdminBundle\Route\RouteCollection;
-use Sonata\AdminBundle\Show\ShowMapper;
-use Sonata\CoreBundle\Validator\ErrorElement;
+use Sonata\AdminBundle\Admin\Extension\AlterNewInstanceInterface;
+use Sonata\AdminBundle\Admin\Extension\ConfigureDatagridFieldsInterface;
+use Sonata\AdminBundle\Admin\Extension\ConfigureFormFieldsInterface;
+use Sonata\AdminBundle\Admin\Extension\ConfigureListFieldsInterface;
+use Sonata\AdminBundle\Admin\Extension\ConfigureQueryInterface;
+use Sonata\AdminBundle\Admin\Extension\ConfigureRoutesInterface;
+use Sonata\AdminBundle\Admin\Extension\ConfigureShowFieldsInterface;
+use Sonata\AdminBundle\Admin\Extension\ConfigureTabMenuInterface;
+use Sonata\AdminBundle\Admin\Extension\GetPersistentParametersInterface;
+use Sonata\AdminBundle\Admin\Extension\PostPersistInterface;
+use Sonata\AdminBundle\Admin\Extension\PostRemoveInterface;
+use Sonata\AdminBundle\Admin\Extension\PostUpdateInterface;
+use Sonata\AdminBundle\Admin\Extension\PrePersistInterface;
+use Sonata\AdminBundle\Admin\Extension\PreRemoveInterface;
+use Sonata\AdminBundle\Admin\Extension\PreUpdateInterface;
+use Sonata\AdminBundle\Admin\Extension\ValidateInterface;
+
+@trigger_error(sprintf(
+    '"%s" is deprecated since 3.x and will be removed in 4.0.'
+            .'Implement single interfaces from "Sonata\AdminBundle\Admin\Extension" namespace.',
```

Comment:
the indentation looks weird and is there a space missing between `... 4.0.` and `Implement...`?

Reply:
yes, space is missing.

Reply:
fixed.